services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -469,6 +468,197 @@ public ConnectedSubnetsWebResponse getConnectedSubnets(String projectId, String
+                }
+                String Des = internalRoutingRule.getDestination();
+                String Nexthop = internalRoutingRule.getNextHopIp();
+
+                updateRoutes.add(internalRoutingRule);
+                hostRouteToSubnet.add(new HostRoute(){{setNexthop(Nexthop);setDestination(Des);}});
+
+            } else { // could find:
+                // compare which one cover the other one
+                if (isNeutronOrVPCLevelRoutingRule) { // VPC/Neutron
+                    String[] existDes = existRoute.getDestination().split("\\/");
+                    String existBitmask = existDes[1];
+                    int existBitmaskInt = Integer.parseInt(existBitmask);
+                    if (newBitmaskInt <= existBitmaskInt) { // new routing rule bitmask is smaller or equal than old one, drop it
+                        continue;
+                    } else { // new routing rule bitmask is larger than old one, Create new rule with low priority
+
+                        InternalRoutingRule internalRoutingRule = constructNewInternalRoutingRule(OperationType.CREATE, RoutingRuleType.DEFAULT, existRoute, newRouteRequest);
+
+                        updateRoutes.add(internalRoutingRule);
+                        hostRouteToSubnet.add(new HostRoute(){{setNexthop(internalRoutingRule.getNextHopIp());setDestination(internalRoutingRule.getDestination());}});
+
+                    }
+                } else { // Subnet
+                    // new routing rule update old one without checking bitmask
+                    InternalRoutingRule internalRoutingRule = constructNewInternalRoutingRule(OperationType.UPDATE, RoutingRuleType.STATIC, existRoute, newRouteRequest);
+
+                    updateRoutes.add(internalRoutingRule);
+                    hostRouteToSubnet.add(new HostRoute(){{setNexthop(internalRoutingRule.getNextHopIp());setDestination(internalRoutingRule.getDestination());}});
+
+                    existRoutesMap.remove(newNetworkIP);
+                }
+
+            }
+
+        }
+
+        for (Map.Entry<String, RouteEntry> existRouteEntry : existRoutesMap.entrySet()) {
+            RouteEntry existRoute = (RouteEntry)existRouteEntry.getValue();
+            InternalRoutingRule internalRoutingRule = constructNewInternalRoutingRule(OperationType.DELETE, RoutingRuleType.DEFAULT, existRoute, null);
+
+            updateRoutes.add(internalRoutingRule);
+
+        }
+
+        // construct List<InternalSubnetRoutingTable>
+        List<InternalSubnetRoutingTable> subnetRoutingTables = new ArrayList<>();
+
+        InternalSubnetRoutingTable internalSubnetRoutingTable = new InternalSubnetRoutingTable();
+        if (owner != null) {
+            internalSubnetRoutingTable.setSubnetId(owner);
+        } else {
+            internalSubnetRoutingTable.setSubnetId(existRouteTable.getOwner());
+        }
```

Comment:
If `owner == null` that's mean we cannot find routetable right? 
then why we still need to set owner to `internalSubnetRoutingTable`?

Reply:
Actually the owner here is not possible to be null, I will change the logic